lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -64,6 +66,15 @@ def setup_hosts
       end
     end
 
+    def effectively_ssl?
+      return @ssl unless @ssl.nil?
+
+      hosts = Array(@hosts)
+      return nil if hosts.nil? || hosts.empty?
```

Comment:
`effectively_ssl?` being a predicate I'd rather only return true/false here:
```suggestion
      return false if hosts.nil? || hosts.empty?
```